SeQuant/core/type_traits.hpp

```diff
@@ -0,0 +1,47 @@
+//
+// Created by Robert Adam on 8/23/23.
+//
+
+#ifndef SEQUANT_TYPETRAITS_HPP
+#define SEQUANT_TYPETRAITS_HPP
+
+#include <type_traits>
+
+namespace sequant {
+
+// Make remove_cvref available also in pre-C++20 code
+#if __cplusplus >= 202002L
+template <typename T>
+using remove_cvref = std::remove_cvref<T>;
+
+using std::remove_cvref_t;
+#else
+template <class T>
+struct remove_cvref {
+  using type = std::remove_cv_t<::std::remove_reference_t<T>>;
+};
+
+template <typename T>
+using remove_cvref_t = typename remove_cvref<T>::type;
+#endif
+
+/// Checks whether \c T is a \c Base (is either the same class or a sub-class
+/// ignoring CV and reference qualifiers
+template <typename Base, typename T>
+using is_a = std::is_base_of<remove_cvref_t<Base>, remove_cvref_t<T>>;
+template <typename Base, typename T>
+constexpr bool is_a_v = is_a<Base, T>::value;
+
+/// Checks whether \c T and \c U are the same type, ignoring any CV and
+/// reference qualifiers
+template <typename T, typename U>
+struct is
```

Comment:
ultra-short names scare me ... how about `sequent::meta_is_same{,_v}` for this? This is basically a slightly improved version of `std::is_same{,_v}`, so makes sense to keep the name.

in same vein: `is_base_of{,_v}` preferred to `is_a{,_v}`

Reply:
I shied away from using longer names as they are essentially exclusively used in nested template specifiers and these get immensely long already...
But if you prefer, I can definitely adapt the naming and try my best at formatting the resulting template code.